### ISSUE TITLE
[a11y-defect-1]: ❗️On focus, button text doesn't meet AA compliance for color contrast

### ISSUE THREAD

**Issue by coforma-jamie** [department-of-veterans-affairs/va.gov-team#64149]
### Point of contact

Jamie Klenetsky Fay @coforma-jamie 

### Severity level

2, Serious. Should be fixed in 1-2 sprints post-launch.

### Details

This is an issue on the final page of the form, the "review and submit" screen.

Give the accordion and edit `<button>` keyboard focus, then use spacebar to activate them. The button background color turns dark blue, and the text is still black, which fails color contrast. This only happens when the button has focus already and you're trying to activate it - you've focused on the accordion heading and want to open it, for example.

Note that this does not happen when using enter to activate the buttons, only spacebar.
![Accordion background dark blue on spacebar activation](https://github.com/department-of-veterans-affairs/va.gov-team/assets/135633989/f856214b-4ade-4aa6-b194-f2bfa97c0069)

![Edit button background dark blue on spacebar activation](https://github.com/department-of-veterans-affairs/va.gov-team/assets/135633989/2569fbff-aa79-41bc-a3e6-f93009063404)

### Reproduction steps

1. Fill out [the form](https://staging.va.gov/careers-employment/vocational-rehabilitation/apply-vre-form-28-1900/introduction) - you'll land on the final page, the review and submit screen.
2. Navigate using the keyboard
3. Give the accordion heading or Edit button focus
4. Hold down spacebar. The background color of the button will turn dark blue.

### Proposed solution or next steps

You'll want to compare your implementation of the accordion and button components to those in the design system, and align yours to those. The DST examples have the proper behavior:
- [Accordion component](https://design.va.gov/components/accordion) / [Storybook](https://design.va.gov/storybook/?path=/docs/uswds-va-accordion--default)
- [Button component](https://design.va.gov/components/button/) / [Storybook](https://design.va.gov/storybook/?path=/docs/uswds-va-button--primary)

See [this Slack thread](https://dsva.slack.com/archives/C01DBGX4P45/p1692968513822459) with the DST - "That is an accordion using the older markup, not an accordion web component."

### References, articles, or WCAG support

1. [WCAG Success Criteria 1.4.3: Contrast (Minimum)](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html)
2. [VA's standards, and how to test for color contrast](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/prepare-for-an-accessibility-staging-review#Prepareforanaccessibilitystagingreview-Useofcolorandcolorcontrastuse-of-color)

### Type of issue
- [ ] Axe-core
- [ ] Screenreader | assistive tech & device support
- [X] Keyboard
- [ ] Zoom
- [X] Color, typography, & visual elements
- [ ] Components and pattern standards
- [ ] Content organization
- [X] Focus
- [X] Headings
- [ ] Content style guide
- [ ] Link and button labels
- [ ] Semantic HTML
- [ ] Markup and metadata
- [ ] Non-text content and media
- [ ] Overall user experience
- [ ] Cognitive | user flows & navigating
- [ ] URL standards
- [ ] Something else

**Comment by zack** [comment]
Hi there, here's what I've found after a little research:

The relevant components on this page are not following modern guidelines. They're using uswds directly, when that is not the guidance. They should be using Foundation or design system components. Updating the edit button is very simply, updating the accordian will be some more work.

If we want to really zoom on this, we can make the fully correct update to the edit button and then hack a fix to the accordion background, **but the correct solution to this is to rearchitect the review page to use the correct accordion component**. I think this will be a _moderate_ lift. Hacking in a better background color is a trivial lift.

**Comment by zack** [comment]
Confirmed we're going to fix everything in this ticket.

**Comment by coforma-jamie** [comment]
That's great @zack, thanks for the update!

**Comment by zack** [comment]
@coforma-jamie it's now open to team review! :) not sure if you like to get in the "internal" review or the "external" review, but have at it!

**Comment by coforma-jamie** [comment]
Woohoo! I'll review it wherever you point me :)

**Comment by coforma-jamie** [comment]
Action items:

- [ ] @coforma-jamie discuss github PR access issue with other a11y specialists
- [ ] @zack install company VPN
- [ ] @zack and @coforma-jamie test sharing files via company VPN

**Comment by coforma-jamie** [comment]
Another a11y specialist with SOCKS access (I'll work on that!) took me through the form and YES, it looks like the issue has been fixed! :)

**Comment by zack** [comment]
merged

**Comment by amponce** [comment]
Hey @zack, I just noticed your update to the edit button. On the financial status report, our review page has a unique layout where items are edited by section. We currently use the .edit-btn class to hide the default buttons. Could we reintroduce the .edit-btn class name into the edit button?

I'm also concerned that your change might affect other a few teams who target that button class in their Cypress tests. Also there's a platform reference to that class in [ ArrayField.jsx](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/platform/forms-system/src/js/review/ArrayField.jsx) 

Thanks!

**Comment by Mottie** [comment]
Yes, I have noticed some focus management has stopped working on the review & submit page; as well as editing content on another page and returning to the review & submit page collapses all the accordions. Hopefully we can revert this change until these issues can be resolved

**Comment by zack** [comment]
Hi, yes we can reintroduce the class if that's necessary. It would seem to me like it would be better not to rely on it (and in fact many tests were updated not to need it), but it's your page! @Mottie, to his credit, suggested that we reintroduce it, but after attempting to find all of the places in the codebase where the class is used, I determined that it was not necessary. The conversation and my findings are in the comment thread [here](https://github.com/department-of-veterans-affairs/vets-website/pull/25744#discussion_r1330661069). Apparently I was mistaken!

**Comment by tblackwe** [comment]
@va-albers do you know who we need to organize to resolve this? This code got reviewed by multiple teams, but some cases were missed in the testing. We can revert the change (which will require dozens of reviews again, or an override of approval process), or we can address the new issues and leave the fix in. 

Once these issues are resolved I'd like to have a post mortem to better understand how to coordinate large scale fixes.

**Comment by tblackwe** [comment]
@Mottie to understand reverting vs fixing, we need to understand the severity of the issues you are seeing. Are you able to provide some detail on what you are experiencing?

**Comment by Mottie** [comment]
I admit that I reviewed the code, but I didn't test anything locally. The
biggest problem is that our app allows editing content on the original
page, and after completing edits, we return the Veteran to the review and
submit page; but upon return, all of the accordions are collapsed. And the
page that was being edited is hidden. We had it set up to return focus to
the original edit button as well.

Additionally, the edit button styling is not aligned, the update page
button is full width, and the change in error styling of the accordions may
not have been discussed with the forms team.

We may not need to fully revert these changes, but we should work on fixing
these issues soon

**Comment by zack** [comment]
Hey @Mottie, I appreciate all your work on this. Your review on the code was incredibly helpful and detailed, so I don't want you to feel like _we_ feel like you didn't review it well!

I'm going to take these issues one at a time here:

> The biggest problem is that our app allows editing content on the original page, and after completing edits, we return the Veteran to the review and submit page; but upon return, all of the accordions are collapsed. And the page that was being edited is hidden. We had it set up to return focus to the original edit button as well.

I didn't even realize this was a required feature. :( I'm afraid this is going to be a pretty big change to the final code since I'm _guessing_ it's going to mean reimplementing all of the redux store code. But I could be wrong on that, I haven't looked at the code in.. well, months.

> Additionally, the edit button styling is not aligned, the update page button is full width

Which one? This one is what's in the video review on the PR and I think it perfectly duplicates the old style? Am I missing something?

<img width="320" alt="image" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/2482403/9ae0b766-f054-464a-88ca-14c20e9629a8">

> and the change in error styling of the accordions may not have been discussed with the forms team.

Finally an easy one. :) I discussed this with the forms team during development and the new style is signed off on for the time being. Plus they have a ticket open to implement the red background again. [Here](https://dsva.slack.com/archives/C01DBGX4P45/p1695826830751769)'s the discussion and [here](https://dsva.slack.com/archives/CBU0KDSB1/p1695827697238279) is the issue.
